test/test_torch.py

```diff
@@ -10394,6 +10349,38 @@ def run_test(tensor_dims, some):
         for tensor_dims, some in product(tensor_dims_list, [True, False]):
             run_test(tensor_dims, some)
 
+    def _test_median(self, device, dtype, inputs):
```

Comment:
This test is OK but I was expected a test that compares median and nanmedian with their NumPy counterparts on a variety of inputs. We're divergent on NumPy for some extremal inputs, like tensors with zero elements and empty tensors, but those divergences can be tested for explicitly in a separate, handwritten test. 

In particular, I was thinking:

- one test with only real values where median and nanmedian are compared with each other (scalar, 1D, 2D, and 3D tensors with different dim and keepdim values tested)
- one test comparing with NumPy on only real-valued tensors and then tensors with some NaN elements (same tensors as before but now some values are NaN at random)
- an "extremal values" test with empty tensors, tensors with zero elements and tensors where some reduced rows consist entirely of NaNs